### PR TITLE
[SYCL-MLIR][polygeist][fix] This patch moves some code around and modifies CMakefiles to build polygeist with shared libraries, which should reduce the time to build.

### PR DESCRIPTION
**Pull request by arnamoy10**

### CODE REVIEW

**Review by Naghasan — APPROVED** [review]
LGTM

File: polygeist/lib/polygeist/Ops.cpp [hunk 1]
```diff
@@ -191,7 +191,37 @@ void BarrierOp::getEffects(
     return;
 }
 
-bool isReadNone(Operation *op);
+// bool isReadNone(Operation *op);
```
Comment (whitneywhtsang) [inline polygeist/lib/polygeist/Ops.cpp:194]: remove

**Comment by etiotto** [comment]
Would it be possible to build the project with shared library by default during the continuous integration? Currently the project is built with static libraries, then our test are run. I think it's better to build with shared library now that this patch fixes that problem.

File: polygeist/lib/polygeist/CMakeLists.txt [hunk 2]
```diff
@@ -11,5 +11,10 @@ MLIRPolygeistOpsIncGen
 LINK_LIBS PUBLIC
 MLIRIR
 MLIRMemRefDialect
+MLIRAffineDialect
```
Comment (etiotto) [inline polygeist/lib/polygeist/CMakeLists.txt:14]: Can you pls put this list in alphabetical order.

Comment (arnamoy10) [reply polygeist/lib/polygeist/CMakeLists.txt:12]: Done

File: polygeist/lib/polygeist/Ops.cpp [hunk 3]
```diff
@@ -191,7 +191,37 @@ void BarrierOp::getEffects(
...
+bool isReadNone(Operation *op) {
+  bool hasRecursiveEffects = op->hasTrait<OpTrait::HasRecursiveSideEffects>();
+  if (hasRecursiveEffects) {
+    for (Region &region : op->getRegions()) {
+      for (auto &block : region) {
```
[the remark this answers is not in the document]
Comment (arnamoy10) [reply polygeist/lib/polygeist/Ops.cpp]: Done

File: polygeist/lib/polygeist/Passes/CMakeLists.txt [hunk 4]
```diff
@@ -37,5 +37,8 @@ add_mlir_dialect_library(MLIRPolygeistTransforms
   MLIRSideEffectInterfaces
   MLIRSCFToControlFlow
   MLIRSYCLDialect
+  MLIRNVVMDialect
```
[the remark this answers is not in the document]
Comment (arnamoy10) [reply polygeist/lib/polygeist/Passes/CMakeLists.txt]: Done

**Pull request merged**